homeassistant/components/sonos/speaker.py

```diff
@@ -840,11 +841,13 @@ def update_media(self, event: SonosEvent | None = None) -> None:
                 self.media.artist = track_info.get("artist")
                 self.media.album_name = track_info.get("album")
                 self.media.title = track_info.get("title")
+                self.media.image_url = track_info.get("album_art")
+                playlist_position = int(track_info.get("playlist_position"))  # type: ignore
+                if playlist_position > 0:
+                    self.media.queue_position = playlist_position - 1
 
                 if music_source == MUSIC_SRC_RADIO:
                     self.update_media_radio(variables)
-                else:
-                    self.update_media_music(track_info)
```

Comment:
Was this un-needed or broken?

Reply:
The attributes it updated were sometimes available on radio stations, too. Moved the contents of that method inline.

Reply:
Actually, let's revert this. It's not necessary for the fixes and may change behavior slightly.